test/test_tensorboard.py

```diff
@@ -0,0 +1,454 @@
+
+
+class SummaryWriterTest(unittest.TestCase):
+    def test_summary_writer_ctx(self):
+        # after using a SummaryWriter as a ctx it should be closed
+        with SummaryWriter(filename_suffix='.test') as writer:
+            writer.add_scalar('test', 1)
+        assert writer.file_writer is None
+
+    def test_summary_writer_close(self):
+        # Opening and closing SummaryWriter a lot should not run into
+        # OSError: [Errno 24] Too many open files
+        passed = True
+        try:
+            writer = SummaryWriter()
+            writer.close()
+        except OSError:
+            passed = False
+
+        assert passed
+
+    def test_windowsPath(self):
+        dummyPath = "C:\\Downloads\\fjoweifj02utj43tj430"
+        with SummaryWriter(dummyPath) as writer:
+            writer.add_scalar('test', 1)
+        import shutil
+        shutil.rmtree(dummyPath)
+
+    def test_pathlib(self):
+        import sys
+        if sys.version_info.major == 2:
+            import pathlib2 as pathlib
+        else:
+            import pathlib
+        p = pathlib.Path('./pathlibtest')
+        with SummaryWriter(p) as writer:
+            writer.add_scalar('test', 1)
+        import shutil
+        shutil.rmtree(str(p))
+
+
+class EmbeddingTest(unittest.TestCase):
+    def test_embedding(self):
+        w = SummaryWriter()
+        all_features = torch.Tensor([[1, 2, 3], [5, 4, 1], [3, 7, 7]])
+        all_labels = torch.Tensor([33, 44, 55])
+        all_images = torch.zeros(3, 3, 5, 5)
+
+        w.add_embedding(all_features,
+                        metadata=all_labels,
+                        label_img=all_images,
+                        global_step=2)
+
+        dataset_label = ['test'] * 2 + ['train'] * 2
+        all_labels = list(zip(all_labels, dataset_label))
+        w.add_embedding(all_features,
+                        metadata=all_labels,
+                        label_img=all_images,
+                        metadata_header=['digit', 'dataset'],
+                        global_step=2)
+        # assert...
+
+    def test_embedding_64(self):
+        w = SummaryWriter()
+        all_features = torch.Tensor([[1, 2, 3], [5, 4, 1], [3, 7, 7]])
+        all_labels = torch.Tensor([33, 44, 55])
+        all_images = torch.zeros((3, 3, 5, 5), dtype=torch.float64)
+
+        w.add_embedding(all_features,
+                        metadata=all_labels,
+                        label_img=all_images,
+                        global_step=2)
+
+        dataset_label = ['test'] * 2 + ['train'] * 2
+        all_labels = list(zip(all_labels, dataset_label))
+        w.add_embedding(all_features,
+                        metadata=all_labels,
+                        label_img=all_images,
+                        metadata_header=['digit', 'dataset'],
+                        global_step=2)
+np.random.seed(0)
+# compare_proto = write_proto  # massive update expect
+
+
+class SummaryTest(unittest.TestCase):
```

Comment:
Please change the class to be `class TestTensorBoardSummary(unittest.TestCase):`